queue/src/fiskaltrust.Middleware.Contracts/RequestCommands/ZeroReceiptCommand.cs

```diff
@@ -50,15 +61,31 @@ public override async Task<RequestCommandResponse> ExecuteAsync(ftQueue queue, R
             var sentReceipts = new List<string>();
             var signatures = new List<SignaturItem>();
 
+            var succeeded = true;
             if (_resendFailedReceipts)
             {
-                await ResendFailedReceiptsAsync(iQueue, queue, sentReceipts, signatures).ConfigureAwait(false);
+                succeeded = await ResendFailedReceiptsAsync(iQueue, queue, queueItem, sentReceipts, signatures).ConfigureAwait(false);
+            }
+
+            var resent = $"Resent {sentReceipts.Count()} receipts that have been stored between {iQueue.SSCDFailMoment:G} and {DateTime.UtcNow:G}.";
+
+            string caption;
+            if (succeeded && signingAvail)
+            {
+                _logger.LogInformation($"Successfully closed failed-mode. {resent} ");
+                caption = $"Restored connection to fiscalization service at {DateTime.UtcNow:G}.";
+                iQueue.SSCDFailCount = 0;
+                iQueue.SSCDFailMoment = null;
+                iQueue.SSCDFailQueueItemId = null;
+            }
+            else
+            {
+                receiptResponse.ftState = _countryBaseState | 2;
+                caption = $"Could not restore connection to fiscalization service at {DateTime.UtcNow:G}.";
             }
             receiptResponse.ftStateData = JsonConvert.SerializeObject(new { SentReceipts = sentReceipts });
-            _logger.LogInformation($"Successfully closed failed-mode, resent {sentReceipts.Count()} receipts that have been stored between {iQueue.SSCDFailMoment:G} and {DateTime.UtcNow:G}.");
 
-            var caption = $"Restored connection to fiscalization service at {DateTime.UtcNow:G}.";
-            var data = $"{iQueue.SSCDFailCount} receipts from the timeframe between {iQueue.SSCDFailMoment:G} and {DateTime.UtcNow:G} have been re-processed at the fiscalization service.";
+            var data = $"{sentReceipts.Count()} receipts from the timeframe between {iQueue.SSCDFailMoment:G} and {DateTime.UtcNow:G} have been re-processed at the fiscalization service.";
```

Comment:
@TSchmiedlechner or @StefanKert any thoughts on that?

Reply:
Hmm not sure if I get this right, but some thoughts on the topic: 
1. Generally, we recommend people to print all signature items, aside from the ones that are flagged as optional (although this flag so far only exists [in Germany](https://docs.fiskaltrust.cloud/docs/poscreators/middleware-doc/germany/reference-tables/ftsignatureformat)). 
2. We also advice PosCreators to store the ReceiptResponse in their databases in addition to ours, so it would probably make sense to keep them as lightweight as possible
3. If I had to start Germany from scratch, I'd try to think of another way to handle the information we need for the DSFinV-K than putting it all in the Signatures 😅

Reply:
Oh, and lastly about this zero receipt: I think the reason why this is done like that in Germany and Austria is the legal requirement to log/print the duration in which the SSCD was not available.